### PR TITLE
[Bug] Update observability saved objects to Discover saved objects

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
Updated the objects as logs has been deprecated to Discover saved objects

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/integrations/components/added_integration.tsx [hunk 1]
```diff
@@ -193,8 +193,8 @@ export function AddedIntegration(props: AddedIntegrationProps) {
                     componentConfig={{
                       activeOption: [
                         {
-                          id: data?.references?.[0]?.id,
-                          label: data?.references?.[0]?.name,
+                          id: data?.references?.[0]?.id || '',
+                          label: data?.references?.[0]?.name || '',
```
Comment (ps48) [inline public/components/integrations/components/added_integration.tsx:197]: May be change to colease:
```
id: data?.references?.[0]?.id ?? '',
label: data?.references?.[0]?.name ?? '',

File: server/adaptors/integrations/integrations_builder.ts [hunk 2]
```diff
@@ -198,8 +316,8 @@ export class IntegrationInstanceBuilder {
     if (options.dataSourceMDSId) {
       instance.references = [
         {
-          id: options.dataSourceMDSId,
-          name: options.dataSourceMDSLabel,
+          id: options.dataSourceMDSId || '',
+          name: options.dataSourceMDSLabel || 'Local cluster',
```
Comment (ps48) [inline server/adaptors/integrations/integrations_builder.ts:320]: same change here?

File: server/adaptors/integrations/integrations_builder.ts [hunk 3]
```diff
@@ -157,6 +204,77 @@ export class IntegrationInstanceBuilder {
...
+    if (!dataSource) {
+      return assets;
+    }
+    return assets.map((asset) => {
+      // Check if the asset type is 'index-pattern' or if the title contains 'Timeline'
+      if (
+        asset?.type &&
+        asset?.attributes?.title &&
+        (asset.type === 'index-pattern' ||
+          (asset.type === 'visualization' && asset.attributes.title.includes('Timeline')))
```
Comment (ps48) [inline server/adaptors/integrations/integrations_builder.ts:219]: `asset.attributes.title.includes('Timeline')))` May not be the right check. 
Should check if  `asset.attributes.visState.type ==="timelion"`

File: server/adaptors/integrations/integrations_builder.ts [hunk 4]
```diff
@@ -157,6 +201,71 @@ export class IntegrationInstanceBuilder {
...
+            type: 'data-source',
+          });
+        }
+      }
+
+      if (asset.type === 'search') {
+        if (
+          asset.attributes &&
+          asset.attributes.kibanaSavedObjectMeta &&
+          asset.attributes.kibanaSavedObjectMeta.searchSourceJSON
```
Comment (ps48) [inline server/adaptors/integrations/integrations_builder.ts:244]: May be just check for this:
```
 asset?.attributes?.kibanaSavedObjectMeta?.searchSourceJSON

File: server/adaptors/integrations/integrations_builder.ts [hunk 5]
```diff
@@ -157,6 +201,63 @@ export class IntegrationInstanceBuilder {
...
+    dataSourceMDSLabel?: string
+  ): SavedObject[] {
+    if (!dataSource) {
+      return assets;
+    }
+    return assets.map((asset) => {
+      // Check if the asset type is 'index-pattern' or if the title contains 'Timeline' visualization
+      if (
+        asset?.type &&
+        asset?.attributes?.title &&
```
Comment (ps48) [inline server/adaptors/integrations/integrations_builder.ts:217]: Can remove the title check here

**Review by TackAdam — APPROVED** [review]

**Review by RyanL1997 — APPROVED** [review]

**Pull request merged**